### PR TITLE
Add types for connect-route

### PR DESCRIPTION
**Pull request by cthompson92**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [ ] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:

- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
## Formatting

The following files are not formatted:
1. types/connect-route/package.json

Consider running `pnpm dprint fmt` on these files to make review easier.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 6f0e5b5181a724097efce1b0e42c03ddbb96fb9d
</p>

File: types/connect-route/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,122 @@
+import { type NextFunction, type NextHandleFunction } from "connect";
+import * as connect from "connect";
+import { type ServerResponse } from "http";
+
+declare namespace ConnectRoute {
+    class IncomingMessage extends connect.IncomingMessage {
```
Comment (sheetalkamat) [inline types/connect-route/index.d.ts:6]: The class declaration cannot be in this namespace as it can only contain types but this is a class

**Review by sheetalkamat — CHANGES_REQUESTED** [review]

File: types/connect-route/index.d.ts (hunk 1 above)
Comment (cthompson92) [reply types/connect-route/index.d.ts]: I see, I went ahead and moved the class definition into a `declare class` outside of the namespace. It still seems to compile and test, so let me know if there's anything else with this class definition.

**Review by sheetalkamat — APPROVED** [review]

**Comment by cthompson92** [comment]
Ready to merge

**Pull request merged**